src/main/java/com/algorand/algosdk/templates/ContractTemplate.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.algorand.algosdk.templates;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+
+import com.algorand.algosdk.crypto.Address;
+import com.algorand.algosdk.crypto.LogicsigSignature;
+import com.algorand.algosdk.util.Encoder;
+
+public class ContractTemplate {
+
+	public enum DataType {
+		BASE64, ADDRESS, INT;
+	}
+
+	public static class ParameterValue {
+		private DataType type;
+		private byte[] byteValue;
+		private int intValue;
+
+		public ParameterValue(DataType dataType, String value) throws NoSuchAlgorithmException {
+			type = dataType;
+
+			if (type == DataType.ADDRESS) {
+				Address addr = new Address(value);
+				byteValue = addr.getBytes();
+			} else {
+				byteValue = value.getBytes();
+			}
+		}
+
+		public ParameterValue(DataType dataType, int value) {
+			type = dataType;
+			intValue = value;
+			if (type != DataType.INT) {
+				throw new RuntimeException("Expecting int type with int value for parameter value.");
+			}
+		}
+
+		public DataType getType() {
+			return type;
+		}
+
+		public int getIntValue() {
+			return intValue;
+		}
+
+		public byte[] getByteValue() {
+			return byteValue;
+		}
+	}
+
+	public String address;
```

Comment:
consider keeping the address as an address object

Reply:
changed! :-)